stellargraph/core/graph.py

```diff
@@ -755,13 +780,19 @@ def node_features(self, nodes=None, node_type=None):
         """
         Get the numeric feature vectors for the specified nodes or node type.
 
-        If ``nodes`` is not specified, this returns all features of the specified ``node_type``,
-        where the rows are ordered the same as ``self.nodes(node_type=node_type)``.
-
-        At least one of ``nodes`` and ``node_type`` must be passed. If ``nodes`` is passed without
-        specifying ``node_type``, the node type of ``nodes`` will be inferred (passing ``node_type``
-        in addition to ``nodes`` will therefore be faster).
+        At least, one of the following conditions must be true:
 
+        - ``node_type`` is passed, in which case ``graph.node_features(node_type=some_type)``
+          returns the features for all nodes of type ``some_type``, in the same order as
+          ``graph.nodes(node_type=some_type)``.
+        - the graph has only one node type, in which case ``graph.node_features()`` returns the
+          features for all nodes (equivalent to the previous case with ``some_type`` set to the
+          single node type of ``graph``)
+        - ``nodes`` is passed, in which case ``graph.node_features(nodes=some_node_ids)`` returns
+          the features for each node in ``some_node_ids`` in the same order as that input. Every
+          node in ``nodes`` must have the same type. If ``nodes`` is passed without specifying
+          ``node_type``, the node type of ``nodes`` will be inferred (passing ``node_type`` in
+          addition to ``nodes`` will therefore be faster).
```

Comment:
This description does make sense to me after reading carefully, but I wonder if it might be easier to parse if homogeneous vs heterogeneous graphs were explained separately? 

For instance, I think it would be nice if the user didn't need to care about the existence of the `node_type` parameter at all if their graph was homogeneous. Maybe something like:
```
For graphs with a single node type, use ``graph.node_features()`` or ``graph.node_features(nodes=some_node_ids)``.

For graphs with multiple node types, the ``node_type`` parameter should be provided:
- ``graph.node_features(node_type=some_type)`` to retrieve features of all nodes of type ``some_type``
- ``graph.node_features(nodes=some_node_ids, node_type=some_type)`` to retrieve features for each node in ``some_node_ids`` which should all be of type ``some_type``. 

The node type can be inferred even for graphs with multiple node types by using ``graph.node_features(nodes=some_node_ids)`` as long as all nodes in ``some_node_ids`` are of a single type, but will be slower than providing the node type explicitly.
```

Reply:
I like it. I went with something slightly different: a list for each graph kind, with 2 and 3 elements respectively. What do you think?

Reply:
Nice! Yeah having a list for both looks good to me